imap_processing/tests/ultra/unit/conftest.py

```diff
@@ -129,11 +134,21 @@ def decom_test_data(request, xtce_path):
     filename = request.param["filename"]
     ccsds_path = imap_module_directory / "tests" / "ultra" / "data" / "l0" / filename
 
-    packets = decom.decom_packets(ccsds_path, xtce_path)
-    grouped_data = group_by_apid(packets)
+    datasets_by_apid = packet_file_to_datasets(ccsds_path, xtce_path)
+
+    strategy_dict = {
+        ULTRA_TOF.apid[0]: process_ultra_tof,
+        ULTRA_EVENTS.apid[0]: process_ultra_events,
+        ULTRA_RATES.apid[0]: process_ultra_rates,
+        ULTRA_TOF.apid[1]: process_ultra_tof,
+        ULTRA_EVENTS.apid[1]: process_ultra_events,
+        ULTRA_RATES.apid[1]: process_ultra_rates,
+    }
 
-    data_packet_list = process_ultra_apids(grouped_data[apid], apid)
-    return data_packet_list, packets
+    process_function = strategy_dict.get(apid, lambda *args: False)
+    data_packet_xarray = process_function(datasets_by_apid[apid])
```

Comment:
Do you need to iterate over apid?

```python
for apid, ds in datasets_by_apid.items():
```

Edit: I later saw that you are getting a specific apid from `request.param`. Ignore this.